specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2016-10-01/keyvault.json

```diff
@@ -573,6 +573,10 @@
             "enum": [
               "resourceType eq 'Microsoft.KeyVault/vaults'"
             ],
+            "x-ms-enum": {
+              "name": "VaultListFilterTypes",
+              "modelAsString": false
+            },
```

Comment:
You should also use the "values" property to give this a better friendly name for code generation. Same as the ResourceManagerApiVersions enum below. For date-based versions we prefer "V2025_11_01". For this expression...I'm not sure. Feel free to come up with something.

Reply:
updated